slides/D-trait-objects-patterns.md

```diff
@@ -0,0 +1,824 @@
+---
+theme: default
+class: text-center
+highlighter: shiki
+lineNumbers: true
+info: "Rust - D: Trait objects and Rust patterns"
+drawings:
+  persist: false
+fonts:
+  mono: Fira Mono
+layout: cover
+title: 'Rust - D: Trait objects and Rust patterns'
+---
+# Rust programming
+Module D: Trait objects and Rust patterns
+<!-- Start with welcome, students entering -->
+
+---
+layout: cover
+---
+# In this module
+
+- Introducing dynamic dispatch
+- Work with design pattern in Rust
+- Common anti-patterns to avoid
+
+---
+layout: default
+---
+# Learning objectives
+<!-- List this module's learning objectives -->
+- Understand the difference between static and dynamic dispatch
+- Be able to create trait objects
+- Understand the concept of 'object safety'
+- Apply various commonly used design patterns
+- Know which anti-patterns to avoid
+
+---
+layout: cover
+---
+#  Module D
+Idiomatic Rust Patterns
+<!-- Start lecture content here -->
+
+---
+layout: default
+---
+# Content overview
+- Trait objects and dynamic dispatch
+- Rust design patterns
+- The deref polymorphism anti-pattern
+- Project intro
+
+---
+layout: section
+---
+# Trait objects & dynamic dispatch
+
+---
+layout: default
+---
+
+# Trait... Object?
+- We learned about traits in module A3
+- We learned about generics and `monomorphization`
+
```

Comment:
I hope you will explain what you mean with that "monomorphization". I had to Google what it means and then went "oh *that*". But perhaps put that a keyword or illustration of it (e.g. a very silly polymorphic function being called on a i32) too.